overlord/state/taskrunner.go

```diff
@@ -235,9 +248,21 @@ func (r *TaskRunner) Ensure() {
 			// Dependencies still unhandled.
 			continue
 		}
+
+		// skip tasks scheduled for later and also track the earliest one
+		if !t.scheduledTime.IsZero() && ensureTime.Before(t.scheduledTime) {
```

Comment:
Same about not touching internals (`if at := task.AtTime(); ... {` would do here).